stdlib/public/core/SmallString.swift

```diff
@@ -241,22 +241,44 @@ extension _SmallString {
   fileprivate mutating func withMutableCapacity(
     _ f: (UnsafeMutableRawBufferPointer) throws -> Int
   ) rethrows {
-    let len = try withUnsafeMutableBytes(of: &self._storage) {
-      (rawBufPtr: UnsafeMutableRawBufferPointer) -> Int in
-      let len = try f(rawBufPtr)
-      UnsafeMutableRawBufferPointer(
-        rebasing: rawBufPtr[len...]
-      ).initializeMemory(as: UInt8.self, repeating: 0)
-      return len
-    }
-    if len == 0 {
+    let len = try withUnsafeMutableBytes(of: &_storage, f)
+
+    if len <= 0 {
+      _debugPrecondition(len == 0)
```

Comment:
Note: This is effectively the same as a `_precondition` in this context, perhaps with some additional code size cost.

Reply:
Good point.

Reply:
Disassembly indicates the code size added by either `_precondition` or `_debugPrecondition` is equivalent. I left it as `_debugPrecondition`.